library/blivet.py

```diff
@@ -665,73 +665,101 @@ def _get_size(self):
+        parent_device = self._blivet_pool._device
+
+        trim_percent = (1.0 - float(parent_device.free_space / size)) * 100
+        log.debug("size: %s ; %s", size, trim_percent)
+
+        if size > parent_device.free_space:
+            if trim_percent > MAX_TRIM_PERCENT:
+                raise BlivetAnsibleError("specified size for volume '%s' exceeds available space in pool '%s' (%s)" % (size,
+                                                                                                                       parent_device.name,
+                                                                                                                       parent_device.free_space))
+            else:
+                log.info("adjusting %s size from %s to %s to fit in %s free space",
+                         self._volume['name'],
+                         size,
+                         parent_device.free_space,
+                         parent_device.name)
+                return parent_device.free_space
+        return size
+
+    def _create_vdo(self):
+        parent_device = self._blivet_pool._device
+        size = self._get_size()
+
+        pool_size = parent_device.free_space if size > parent_device.free_space else size
+
+        if self._volume['vdo_pool_size']:
+            try:
+                pool_size = self._trim_size(Size(self._volume['vdo_pool_size']))
+            except BlivetAnsibleError:
+                # Literally the same error just more specific description
+                raise BlivetAnsibleError("specified 'vdo_pool_size' for volume '%s' "
+                                         "exceeds available space in pool '%s' (%s)" % (pool_size,
+                                                                                        parent_device.name,
+                                                                                        parent_device.free_space))
+
+        try:
+            vdopool = self._blivet.new_lv(name='vdopool', vdo_pool=True,
+                                          parents=[parent_device], compression=self._volume['compression'],
+                                          deduplication=self._volume['deduplication'],
+                                          size=pool_size)
+        except Exception as e:
+            raise BlivetAnsibleError("failed to set up VDO pool '%s': %s" % (self._volume['name'], str(e)))
+
+        self._blivet.create_device(vdopool)
+
+        return dict(vdo_lv=True, parents=[vdopool])
+
+    def _create_lvmraid(self):
+        parent_device = self._blivet_pool._device
+
+        if self._volume['raid_disks']:
+            pvs = list()
+
+            for path in self._volume['raid_disks']:
+                for pv in parent_device.pvs:
+                    disk_path = pv.path if pv.is_disk else pv.disk.path
+                    # We need to ensure we match '/dev/sda' with 'sda' but not with 'da'
+                    if ('/' + disk_path).endswith('/' + path):
```

Comment:
You can use `os.path.basename` to get name of the device from path, but I would suggest using `resolve_device` to get blivet devices for the `raid_disks` and then compare these to ancestors of the PVs.

Reply:
Fixed.